README.md

```diff
@@ -1,302 +1,227 @@
 
 ## Usage
 
-All the following invocations assume valid credentials are present in the
-environment. The following examples all reference an `INSTANCE_CONNECTION_NAME`,
-which takes the form: `myproject:myregion:myinstance`. To find the
-`INSTANCE_CONNECTION_NAME`, run `gcloud sql instances describe <INSTANCE_NAME>`
-where `INSTANCE_NAME` is the name of the database instance.
+The following examples all reference an `INSTANCE_CONNECTION_NAME`, which takes
+the form: `myproject:myregion:myinstance`.
 
-### TCP socket example
+To find your Cloud SQL instance's `INSTANCE_CONNECTION_NAME`, visit the detail
+page of your Cloud SQL instance in the console, or use `gcloud` with:
 
-``` bash
-# Starts the proxy listening on 127.0.0.1:5432
-cloud_sql_proxy -instances=<INSTANCE_CONNECTION_NAME>=tcp:5432
+```shell
+gcloud sql instances describe <INSTANCE_NAME> --format='value(connectionName)'
 ```
 
-``` bash
-# Starts the proxy listening on port 5432 on *all* interfaces
-cloud_sql_proxy -instances=<INSTANCE_CONNECTION_NAME>=tcp:0.0.0.0:5432
-```
+### Basic Usage
 
-### Unix socket example
+To start the proxy, use:
 
-``` bash
-# The proxy will mount a Unix domain socket at /cloudsql/<INSTANCE_CONNECTION_NAME>
-# Note: The directory specified by `-dir` must exist and the socket file path
-# (i.e., dir plus INSTANCE_CONNECTION_NAME) must be under your platform's
-# limit (typically 108 characters on many Unix systems, but varies by platform).
-cloud_sql_proxy -dir=/cloudsql -instances=<INSTANCE_CONNECTION_NAME>
+```shell
+# starts the proxy listening on localhost with the default database engine port
+# For example:
+#   MySQL      localhost:3306
+#   Postgres   localhost:5432
+#   SQL Server localhost:1433
+./cloudsql-proxy <INSTANCE_CONNECTION_NAME>
 ```
 
-### Private IP example
+The proxy supports multiple instances:
 
+```shell
+./cloudsql-proxy <INSTANCE_CONNECTION_NAME_1> <INSTANCE_CONNECTION_NAME_2>
 ```
-cloud_sql_proxy -instances=<INSTANCE_CONNECTION_NAME>=tcp:5432 -ip_address_types=PRIVATE
-```
-
-In order to connect using Private IP, you must have access through your
-project's VPC. For more details, see [Private IP Requirements][private-ip].
-
-## Credentials
-
-The Cloud SQL proxy uses a Cloud IAM account to authorize connections against a
-Cloud SQL instance. The proxy sources the credentials for these accounts in the
-following order:
 
-1. The `-credential_file` flag
-2. The `-token` flag
-3. The service account key at the path stored in the
-   `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
-4. The gcloud user credentials (set from `gcloud auth login`)
-5. The [Application Default Credentials](https://cloud.google.com/docs/authentication/production)
-
-Note: Any account connecting to a Cloud SQL database will need one of the
-following IAM roles:
+### Configuring Port
 
-- Cloud SQL Client (preferred)
-- Cloud SQL Editor
-- Cloud SQL Admin
+To override the port, use the `--port` flag:
```

Comment:
We should recommend that production environments specify a port to reduce start up time here